src/rebar_prv_dialyzer.erl

```diff
@@ -554,21 +561,32 @@ legacy_warnings(Warnings) ->
             Warnings
     end.
 
-format_warnings(Opts, Output, Warnings) ->
+format_warnings(State, Output, Warnings) ->
+    Opts = rebar_state:opts(State),
     Warnings1 = rebar_dialyzer_format:format_warnings(Opts, Warnings),
     console_warnings(Warnings1),
-    file_warnings(Output, Warnings),
+
+    OutputFormat = get_config(State, output_format, formatted),
+    file_warnings(Output, Warnings, OutputFormat),
+
     length(Warnings).
 
 console_warnings(Warnings) ->
     _ = [?CONSOLE("~ts", [Warning]) || Warning <- Warnings],
     ok.
 
-file_warnings(_, []) ->
+file_warnings(_, [], _) ->
     ok;
-file_warnings(Output, Warnings) ->
-    Warnings1 = [[dialyzer:format_warning(Warning, fullpath), $\n] || Warning <- Warnings],
-    case file:write_file(Output, Warnings1, [append]) of
+file_warnings(Output, Warnings, raw) ->
+    Warnings1 = [[io_lib:format("~tp. \n", [W]) || W <- Warnings]],
```

Comment:
no need for a trailing space after items there if you have a linebreak I believe.